parlai/crowdsourcing/projects/humaneval/single_model_eval/analysis/compile_results.py

```diff
@@ -0,0 +1,341 @@
+# LICENSE file in the root directory of this source tree.
+
+import os
+from collections import defaultdict
+from datetime import datetime
+from typing import Any, Dict
+
+import pandas as pd
+from mephisto.data_model.worker import Worker
+
+import parlai.utils.logging as logging
+from parlai.crowdsourcing.tasks.model_chat.model_chat_blueprint import (
+    BLUEPRINT_TYPE,
+)  # For registering the blueprint
+from parlai.crowdsourcing.utils.acceptability import AcceptabilityChecker
+from parlai.crowdsourcing.utils.analysis import AbstractResultsCompiler
+
+_ = BLUEPRINT_TYPE
+# NOTE: BLUEPRINT_TYPE needs to be imported here to register the blueprint
+
+
+class ModelChatResultsCompiler(AbstractResultsCompiler):
+    """
+    Compile and save results of human+model chats.
+
+    Results will be saved on the level of specific conversations, as well as aggregated
+    up to the level of each worker as a whole.
+    """
+
+    @classmethod
+    def setup_args(cls):
+        parser = super().setup_args()
+        parser.add_argument(
+            '--filter-uniform-hits',
+            action='store_true',
+            help='Filter out any HITs in which the worker\'s annotations were the exact same on each turn of the conversation',
+        )
+        return parser
+
+    def __init__(self, opt: Dict[str, Any]):
+
+        super().__init__(opt)
+
+        # Input args
+        os.makedirs(self.output_folder, exist_ok=True)
+        # TODO: see if this can be moved to the superclass
+        self.filter_uniform_hits = opt['filter_uniform_hits']
+
+        # Save paths
+        self.unacceptable_worker_ids_path = os.path.join(
+            self.output_folder, 'unacceptable_worker_ids.txt'
+        )
+        self.annotation_selection_rate_path = os.path.join(
+            self.output_folder, 'annotation_selection_rates.csv'
+        )
+        self.likert_score_stat_path = os.path.join(
+            self.output_folder, 'likert_score_stats.csv'
+        )
+
+        self.acceptability_checker = AcceptabilityChecker()
+
+    def get_results_path_base(self) -> str:
+        return os.path.join(self.output_folder, 'results')
+        # TODO: see if this can be moved to the superclass
+
+    def compile_results(self) -> pd.DataFrame:
+
+        # Load task data
+        logging.info('Retrieving task data from Mephisto.')
+        task_units_data = self.get_task_data()
+        logging.info(f'Data for {len(task_units_data)} units loaded successfully.')
+
+        num_convos_with_no_save_data = 0
+        num_wrong_status_convos = 0
+        num_complete_convos = 0
+
+        unacceptable_task_units = []
+        unacceptable_worker_ids = []
+        conversation_idx = 0
+        conversation_dfs = []
+
+        for task_unit in task_units_data:
+
+            worker_id = task_unit['worker_id']
+            assignment_id = task_unit['assignment_id']
+
+            # # Determining whether the task unit should be skipped
```

Comment:
Yeah good point - so, my convention has often been to use two hash symbols for a new section of code: i.e., this comment is indicating that the following 100-ish lines deal with checking whether this task should be skipped. But if this isn't clear to others, then maybe something more obvious should be used. Is there a particular notation that you use for this?

Reply:
As an aside, often if I find a chunk of code is complex enough to warrant something like that, it's complex enough to move into a nicely-named helper function. 
```python
if should_skip_unit(...):
  continue
```

I understand that in this scripting setup, this chunk also has some external effects (changing convos counts, doing some data extraction, etc). But in that case, those are critical to their function, and thus the whole code block is doing more than "determining whether the task unit should be skipped". Overall feels a little like an anti-pattern to me.

Reply:
Yes - @JackUrb definitely agreed that the original phrasing of that comment was incomplete. I've just removed that comment entirely to avoid confusion.